File: src/ProjectR.Scripting/content/scripts/spells/lua/ThousandNeedles.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Thousand Needles"
end

function GetDescription()
    return "Wears down the enemies defense."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .6
end

function GetDelay()
  return .3
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 4 * aMD - dDEF  
  d:BuffStat(DEF, -.25)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/ThousandNeedles.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class ThousandNeedles : SpellScriptBase
    {
        public override string Name { get { return "Thousand Needles"; } }
        public override string Description { get { return "Wears down the enemies defense."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 60; } }
        public override double Delay { get { return .3; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 4 * aMD - dDEF;
            BuffStat(Stat.DEF, -.25);
            DealDamage(damage);
        }
    }
}